app/lib/actions/katello/upstream_subscriptions/remove_entitlements.rb

```diff
@@ -3,17 +3,26 @@ module Katello
     module UpstreamSubscriptions
       class RemoveEntitlements < Actions::Base
         def plan(pool_ids = [])
+          organization = Object::Organization.current
+
+        begin
```

Comment:
I think the RestClient::Gone exception shouldn't be reaching up this high for a few reasons. I prefer to see knowledge of our HTTP client (RestClient) limited to the Resource:: namespace. That'll make our lives easier when we inevitably change clients.

The answer to this is to have the .get method itself raise UpstreamConsumerGone similar to this:

https://github.com/Katello/katello/blob/fe7244b506f3439757d821be3516eb2f460ac97a/app/lib/katello/resources/candlepin/upstream_pool.rb#L8-L12